### PR TITLE
feat: add options parameter to DeepLTranslate() to accept additional translation options

### PR DESCRIPTION
**Pull request by ghost**
Adds another (fifth) parameter to DeepLTranslate that lets you pass any custom options through in API requests.

### CODE REVIEW

File: DeepL.gs [hunk 1]
```diff
@@ -75,6 +82,22 @@ function DeepLTranslate(input, sourceLang, targetLang, glossaryId) {
     if (glossaryId) {
         formData['glossary_id'] = glossaryId;
     }
+    if (options) {
+        if (!Array.isArray(options) ||
+            !Object.values(options).every(function(value) {
```
Comment (JanEbbing) [inline DeepL.gs:87]: I don't think this condition is completely correct.
`!Array.isArray(options) || ...` means we throw if `options` is not an array (`[...]`), and then `... || !Object.values.every....` checks that each entry is of length two. I assume this is to support the cell range approach? But I would expect:

- This to throw when we pass a dictionary/an Object with the options (`Arrays.isArray({'tag_handling':'xml'})` is false)
- This to break once we are beyond row 10/column 26 (Cell A10 has length 3, right?)

File: README.md [hunk 2]
```diff
@@ -206,6 +206,34 @@ into B2.
...
+The fifth argument to `DeepLTranslate()` accepts the options specified as a
+range with two columns: the option name and option values. You can specify the 
+options somewhere in the sheet and refer to them in your translations, as shown
+in the following example:
+
+```=DeepLTranslate(A1,,"de",,C2:D4)```
+
+![Translating the second cell](docs/DeepL_Function_Call_Additional_Options.png)
+
+Note that the `source_lang` and `glossary_id` parameters are used in this
```
Comment (JanEbbing) [inline README.md:224]: Nit: Typo
> Note that the `source_lang` and `glossary_id` parameters are **not** used in this

File: README.md [hunk 3]
```diff
@@ -206,6 +206,34 @@ into B2.
...
+example, so they are empty.
+
+If you are translating multiple cells, you may want to make the reference to the
+options absolute (`$C$2:$D$4`).
+
+#### Inline formula options
+
+Google Sheets also allows you to specify ranges in a formula using the
```
Comment (JanEbbing) [inline README.md:232]: Nit: Not sure we want to put this technical background in the README. I would probably rephrase as

> You can also pass the options to the `DeepLTranslate` function directly using the `{opt1, val1; opt2, val2; ..}` syntax, for example:

**Review by JanEbbing — CHANGES_REQUESTED** [review]
Nice, I really like passing options via cells, makes it much more readable and reduces copy/paste.
Requesting changes since I dont think the condition is correct, but I might be wrong/GSheets does something I don't know about before passing options to us

File: DeepL.gs (hunk 1 above)
Comment (daniel-jones-deepl) [reply DeepL.gs:87]: The `{..}` syntax used in Google Sheets is deceptive -- in Apps Script it always seems to be converted to an array of arrays. It is possibly a Google-defined type (maybe CellRange or something) that works like an array.

- Passing `{'tag_handling', 'xml'}` as the argument in the cell formula will still yield an array in the script. 
- Passing `{'tag_handling': 'xml'}` didn't seem to work in my tests; in the script it was passed a string with a value like "#NAME".

The arrays' lengths should correspond to the number of columns (i.e. A, B are two columns). I don't think this changes one multi-character row- or column-encodings are used, but I'll check.

**Review by JanEbbing — APPROVED** [review]

File: DeepL.gs (hunk 1 above)
Comment (JanEbbing) [reply DeepL.gs:87]: Got it, that makes sense.

Comment (daniel-jones-deepl) [reply DeepL.gs:87]: Confirmed it works beyond column AA and row 10.

**Pull request merged**